odoo/tools/query.py

```diff
@@ -221,6 +221,18 @@ def __bool__(self):
         return bool(self.get_ids())
 
     def __len__(self):
+        if self._ids is None and not self.offset:
```

Comment:
It's a limitation of the implementation: the case of having a `SELECT COUNT(*)` with an `OFFSET` is not covered here.  In practice, this is not really an issue, since method `search_count()` should not generate such queries.